source/read/change-streams.txt

```diff
@@ -0,0 +1,268 @@
+
+.. meta::
+   :keywords: watch, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use a **change stream** to monitor real-time
+changes to your data. A change stream is a {+mdb-server+} feature that
+allows your application to subscribe to data changes on a collection, database,
+or deployment.
+
+When using the {+php-library+}, you can call the ``watch()`` method to return an
+instance of ``MongoDB\ChangeStream``. Then, you can iterate through change events stored 
+in the ``MongoDB\ChangeStream`` instance to monitor data changes, such as updates,
+insertions, and deletions.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your PHP application, instantiate a ``MongoDB\Client`` that connects to an Atlas cluster
+and assign the following value to your ``$collection`` variable:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :language: php
+   :dedent:
+   :start-after: start-db-coll
+   :end-before: end-db-coll
+
+.. tip:: 
+
+   To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+   :atlas:`Get Started with Atlas </getting-started>` guide.
+
+Some examples use the ``toJSON()`` function to represent change events, which are BSON 
+documents, as Extended JSON. To use this function, paste the following code into your
+application file:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :language: php
+   :dedent:
+   :start-after: start-to-json
+   :end-before: end-to-json
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call the ``watch()`` method. The instance on which you
+call the ``watch()`` method determines the scope of events that the change
+stream monitors. You can call the ``watch()`` method on instances of the following
+classes:
+
+- ``MongoDB\Client``: Monitor all changes in the MongoDB deployment
+- ``MongoDB\Database``: Monitor changes in all collections in the database
+- ``MongoDB\Collection``: Monitor changes in the collection
+
+The following example opens a change stream on the ``restaurants`` collection
+and outputs changes as they occur:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-open-change-stream
+   :end-before: end-open-change-stream
+   :language: php
+   :dedent:
+
+To begin watching for changes, run the preceding code. Then, in a separate
+application or shell, modify the ``restaurants`` collection. The following
+example updates a document that has a ``name`` field value of ``'Blarney Castle'``:
+
+.. _php-change-stream-update:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-update-for-change-stream
+   :end-before: end-update-for-change-stream
+   :language: php
+   :dedent:
+
+When you update the collection, the change stream application prints the change
```

Comment:
Something that just occurred to me: should this tutorial advise users to run this PHP code through a shell/terminal? Users would likely run into issues trying to execute the `watch()` process through a web server.

I'm curious if there's language from other language tutorials (assuming this was adapted from existing content) we can use.

The most straightforward way to address this might be to consistently refer to "shell process" instead of "application or shell" for all of the code examples (no harm in using that for the update example as well).

Reply:
We use "application" pretty consistently across all docs, but I agree that it makes sense to suggest running the application from the shell. I updated all mentions of running the application on this page to clarify that